### ISSUE TITLE
[HLSL] Remove old resource annotations

### ISSUE THREAD

**Issue by hekota** [llvm/llvm-project#114126]
Once we move UAV, SRV and constant buffers to the LLVM target types we can remove the metadata resource annotations since they are no longer needed.

That means removing all of the code that generates the `!hlsl.uavs`, `!hlsl.srvs` and `hlsl.cbufs` metadata from `CGHLSLRuntime`.

Related task: llvm/llvm-project#104862

**Comment by damyanp** [comment]
Need cbuffers to be implemented before we can do this.
